controller/rest/longpoll.go

```diff
@@ -52,54 +58,77 @@ func (j *longpollOnceJob) Poll() (interface{}, time.Duration) {
 	}
 }
 
+type PollWorkItem struct {
+	key  interface{}
+	args interface{}
+	task longpollOnceTask
+}
+
 type longpollOnceMgr struct {
-	mux     sync.Mutex
-	max     int
-	timeout time.Duration
-	linger  time.Duration
-	jobs    map[interface{}]*longpollOnceJob
+	mux       sync.RWMutex
+	poolSize  int
+	timeout   time.Duration
+	linger    time.Duration
+	jobs      map[interface{}]*longpollOnceJob // use as a queue
+	workQueue chan PollWorkItem
```

Comment:
`jobs` and `workQueue` seem to serve similar purposes.  Have you considered replacing `jobs` with `workQueue`?

Reply:
Hi @holyspectral ,
I considered this approach before; however, there's a tricky case where the user continuously sends the same event. In this scenario, the controller needs a way to recognize that it's a duplicate job. This is why we require a map to keep track of which jobs have already been processing.

Reply:
Is it possible for users to NOT send duplicate scan requests?

Reply:
Overall I think the interaction between `jobs` and `workQueue` has to be thought through carefully.  
For example, when call#1 and call#2 calls the same API twice with the same payload, `m.jobs[key]` would be overwritten by the job that comes later.

However, when `pollWorker()` really starts the job, there is no guarantee that which job will be used, so this will make one of the request time out.  

If you want to keep both, you will have to be very careful to deal with this kind of issues.